packages/mapviewer/src/modules/drawing/DrawingModule.vue

```diff
@@ -195,12 +201,26 @@ onBeforeUnmount(() => {
 
     document.removeEventListener('contextmenu', removeLastPointOnRightClick)
     document.removeEventListener('keyup', removeLastPointOnDeleteKeyUp)
+    window.removeEventListener('beforeunload', beforeUnloadHandler)
 
     if (IS_TESTING_WITH_CYPRESS) {
         delete window.drawingLayer
     }
 })
 
+const beforeUnloadHandler = (event) => {
+    if (showNotSharedDrawingWarning.value) {
+        showNotSharedDrawingWarningModal.value = true
+        // This provokes the alert message to appear when trying to close the tab.
+        // During Cypress tests this causes the test to run indefinitely, so to prevent this we skip the alert.
+        if (!IS_TESTING_WITH_CYPRESS) {
+            event.returnValue = true
+            event.preventDefault()
+            return
+        }
+    }
```

Comment:
are we testing the popup in Cypress?
if not, we could entirely skip the change of `showNotSharedDrawingWarningModal` by editing the if case so that
```
if (!IS_TESTING_WITH_CYPRESS && showNotShareDrawingWarning.value) {
    // ...
}
```